futures-channel/src/oneshot.rs

```diff
@@ -237,6 +237,23 @@ impl<T> Inner<T> {
         }
     }
 
+    fn try_recv(&self) -> Result<Option<T>, Canceled> {
+        // If we're complete, either `::close_rx` or `::drop_tx` was called.
+        // We can assume a successful send if data is present.
+        if self.complete.load(SeqCst) {
+            if let Some(mut slot) = self.data.try_lock() {
+                if let Some(data) = slot.take() {
+                    return Ok(Some(data.into()));
+                }
+            }
+            // Should there be a different error value or a panic in the case
```

Comment:
I think this error is fine-- it's the same thing you'd get if you tried calling `recv` after it had already completed successfully.